src/main/java/com/example/decafe/HelloController.java

```diff
@@ -0,0 +1,577 @@
+    public void changePlayAgainBack() throws FileNotFoundException {
+        playAgainImage.setImage(createImage("playAgain.png"));
+    }
+
+    // end screen - change BackToStartMenu Button when mouse entered
+    public void changeBackToStartMenu() throws FileNotFoundException {
+        backToStartImage.setImage(createImage("backToStartMenuBrighter.png"));
+    }
+
+    // end screen - change BackToStartMenu Button when mouse exited
+    public void changeBackToStartMenuBack() throws FileNotFoundException {
+        backToStartImage.setImage(createImage("backToStartMenu.png"));
+    }
+
+    // end screen - change Quit Button when mouse entered
+    public void changeQuitEndScreen() throws FileNotFoundException {
+        quitEndScreenImage.setImage(createImage("quitEndScreenBrighter.png"));
+    }
+
+    // end screen - change Quit Button when mouse exited
+    public void changeQuitEndScreenBack() throws FileNotFoundException {
+        quitEndScreenImage.setImage(createImage("quitEndScreen.png"));
+    }
+
+    // if waiter is near coffee machine, change appearance when clicked
+    public void showCoffee() throws FileNotFoundException {
+        if (waiterImageView.getBoundsInParent().intersects(coffeeMachineImageView.getBoundsInParent())) {
+            Play.getCoffeeMachine().displayProduct(waiterImageView, coffeeMachineImageView, CofiBrew, progressBarCoffee);
+            File f = new File("");
+            String musicFile = f.getAbsolutePath() + File.separator + "src" + File.separator + "main" + File.separator + "resources" + File.separator + "com" + File.separator + "example" + File.separator + "decafe" + File.separator + "test_sound.wav";
+            AudioClip coffeeSound = new AudioClip(new File(musicFile).toURI().toString());
+            //MediaPlayer coffeeSound = new MediaPlayer(sound);
+            coffeeSound.play();
+        }
+    }
+
+    // if waiter is near cake machine, change appearance when clicked
+    public void showCake() throws FileNotFoundException {
+        if (waiterImageView.getBoundsInParent().intersects(cakeMachineImageView.getBoundsInParent())) {
+            Play.getCakeMachine().displayProduct(waiterImageView, cakeMachineImageView, CofiBrew, progressBarCake);
+            File f = new File("");
+            String musicFile = f.getAbsolutePath() + File.separator + "src" + File.separator + "main" + File.separator + "resources" + File.separator + "com" + File.separator + "example" + File.separator + "decafe" + File.separator + "test_sound.wav";
+            AudioClip cakeSound = new AudioClip(new File(musicFile).toURI().toString());
+            //MediaPlayer cakeSound = new MediaPlayer(sound);
+            cakeSound.play();
+        }
+    }
+
+    // if no product is held by waiter
+    public void noProduct() throws FileNotFoundException {
+        if (CofiBrew.getProductInHand().equals("coffee") || CofiBrew.getProductInHand().equals("cake")) {
+            File f = new File("");
```

Comment:
there is another File with the variable name f at line 136, therefore renaming would avoid confusion....